### PR TITLE
fixes playlist blurring

### PR DESCRIPTION
**Pull request by esotrope**
image

### CODE REVIEW

File: webapp/src/app/resource/components/protected-content/protected-content.component.scss [hunk 1]
```diff
@@ -2,8 +2,9 @@
   position: relative;
 
   &.obscured {
-    ::ng-deep > :not(sbdl-preview-login-prompt) {
-      filter: blur(10px);
+    ::ng-deep > *:not(sbdl-preview-login-prompt) {
+      display: block;
```
Comment (esotrope) [inline webapp/src/app/resource/components/protected-content/protected-content.component.scss]: the magic was the filter doesn't apply unless it is display block i guess : /

File: webapp/src/app/resource/components/protected-content/protected-content.component.scss [hunk 2]
```diff
@@ -2,8 +2,9 @@
...
+      filter: blur(1rem);
```
Comment (esotrope) [inline webapp/src/app/resource/components/protected-content/protected-content.component.scss]: increased the blur for less visual tension

**Comment by esotrope** [comment]
Accidentally pushed the fix for 322 (note panel not opening on notepad link from bookmark page) into this PR

File: webapp/src/app/resource/components/actions/actions.component.html [hunk 3]
```diff
@@ -1,9 +1,9 @@
-<sbdl-button-icon *ngIf="isAuthenticated" buttonTitle="Notes" (click)="toggleNotes()">
+<sbdl-button-icon *ngIf="authenticated$ | async" buttonTitle="Notes" (click)="toggleNotes()">
   <sbdl-icon [ngClass]="{'has-notes': hasNotes}" icon="take-note"></sbdl-icon>
 </sbdl-button-icon>
 <sbdl-button-icon buttonTitle="Share" (click)="share()"><i class="far fa-share"></i></sbdl-button-icon>
 <ng-container #shareButton></ng-container>
-<sbdl-button-icon *ngIf="isAuthenticated" buttonTitle="Bookmark" (click)="toggleBookmarked()">
+<sbdl-button-icon *ngIf="authenticated$ | async" buttonTitle="Bookmark" (click)="toggleBookmarked()">
```
Comment (esotrope) [inline webapp/src/app/resource/components/actions/actions.component.html]: one thing nice about doing things the reactive way like this is the angular template language will both perform the subscription and unsubscription for us

File: webapp/src/app/resource/instructional/instructional-resource.component.html [hunk 4]
```diff
@@ -7,9 +7,9 @@
       [readingMode]="readingMode"
       [printingMode]="printingMode"
       [resource]="resource"
-      (readingModeChanged)="readingModeChanged($event)"
-      (printingModeChanged)="printingModeChanged($event)"
-      (notesVisibilityChanged)="notesVisibilityChanged($event)"
+      (readingModeChanged)="readingModeChanged.emit($event)"
+      (printingModeChanged)="printModeChanged.emit($event)"
+      (notesVisibilityChanged)="noteModeChanged.emit($event)"
```
Comment (esotrope) [inline webapp/src/app/resource/instructional/instructional-resource.component.html]: made a design tweak here.
before the state was initialized by the resource-strategy-comp then changes to state were only made locally to this component and never was the state taken from the query params.
Now the resource-strategy-comp subscribes to these events of the resource, updates the url query string, and makes sure any change or initial query string is pushed down as the current state

File: webapp/src/app/resource/resource-content.component.ts [hunk 5]
```diff
@@ -43,25 +42,9 @@ export class ResourceContentComponent implements OnInit {
...
-      this.readingModeChanged.emit(event);
-    }
-
-    emitPrintingModeChanged(event) {
-      this.printingModeChanged.emit(event);
-    }
-
-    emitNotesVisibilityChanged(event) {
-      this.notesVisibilityChanged.emit(event);
-    }
```
Comment (esotrope) [inline webapp/src/app/resource/resource-content.component.ts]: unused

File: webapp/src/app/resource/resource-type-strategy.component.ts [hunk 6]
```diff
@@ -16,50 +31,81 @@ import { StrategyResourceComponent } from './strategy/strategy-resource.componen
...
 
-      const viewContainerRef = this.hostDirective.viewContainerRef;
-      viewContainerRef.clear();
-      const componentRef = viewContainerRef.createComponent(componentFactory);
+    const viewContainerReference = this.hostDirective.viewContainerRef;
 
-      componentRef.instance.resource = data.resource;
-      componentRef.instance.notes = data.notes;
-      this.componentInst = componentRef.instance;
+    combineLatest(
```
Comment (esotrope) [inline webapp/src/app/resource/resource-type-strategy.component.ts]: the new resource-strategy-comp state orchestration

File: webapp/src/app/resource/resource-type-strategy.component.ts [hunk 7]
```diff
@@ -16,50 +31,81 @@ import { StrategyResourceComponent } from './strategy/strategy-resource.componen
...
+      componentReference.instance.showNotes = queryParams[showNotesParameter] === 'true' && authenticated;
     });
+  }
 
-    // TODO: Move into ResourceComponent. Set in OnInit, subscribe in
-    // AfterViewInit. This has been broken by the SSO integration. We now hide
-    // the notes components unless the user is logged in. These initial checks
-    // happen before the ResourceComponent has loaded and so before the user
-    // object is available.
-    this.route.queryParams.subscribe(queryParams => {
```
Comment (esotrope) [inline webapp/src/app/resource/resource-type-strategy.component.ts]: by using combineLatest or just listening to `userService.{user|authenticated}` changes we can make sure we respond to changes in the authenticated state

File: webapp/src/app/resource/resource-type-strategy.component.ts [hunk 8]
```diff
@@ -16,50 +31,81 @@ import { StrategyResourceComponent } from './strategy/strategy-resource.componen
...
 
-      if (queryParams.viewNotes !== undefined) {
-        this.componentInst.notesVisibilityChanged(queryParams.viewNotes);
+  private setQueryParameter(parameterName: string, value: any): void {
+    this.router.navigate(['.'], {
+      relativeTo: this.route,
+      replaceUrl: true,
+      queryParams: {
+        [parameterName]: value
       }
```
Comment (esotrope) [inline webapp/src/app/resource/resource-type-strategy.component.ts]: changes query params in-place without refreshing the page or adding to the browser history

File: webapp/src/app/resource/resource.component.ts [hunk 9]
```diff
@@ -8,37 +8,46 @@ import { User } from '../data/user/user.model';
...
  * Parent class that other resource component classes extend.
  * No @Component attribute here because it is never used directly.
  */
 export class ResourceComponent implements AfterViewInit, OnInit {
 
-  resource: Resource;
-  notesVisible = false;
-  notes: Note[];
+  @Output()
+  readonly readingModeChanged: EventEmitter<boolean> = new EventEmitter();
```
Comment (esotrope) [inline webapp/src/app/resource/resource.component.ts]: now all we need is zen mode : j

**Comment by jdbernard** [comment]
@esotrope Can you take ownership of [DL-273](https://smarterbalanced.atlassian.net/browse/DL-273), move it to "QA Ready," and assign to Nohemi once you've merged this branch and the playlist blur branch? It was hanging on waiting for the fixes you have done.

**Comment by esotrope** [comment]
Sure thing. I'll take it from here : j

**Pull request merged**